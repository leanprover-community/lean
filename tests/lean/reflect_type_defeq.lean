meta def foo (x : reflected ℕ 3) : reflected (list ℕ) [10] :=
x -- ERROR

def synonym (α : Type) : Type := α

def synonym.of {α : Type} (a : α) : synonym α := a

meta instance synonym.reflect {α : Type} [h : has_reflect α] [reflected _ α] :
  has_reflect (synonym α) :=
λ x, show reflected _ (synonym.of x), from `(synonym.of).subst (h x)

run_cmd let x := synonym.of 1 in tactic.trace `(x)
run_cmd let x := 1 in tactic.trace `(x)
