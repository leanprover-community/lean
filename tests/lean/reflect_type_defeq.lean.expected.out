reflect_type_defeq.lean:2:0: error: equation type mismatch, term
  x
has type
  reflected ℕ 3
but is expected to have type
  reflected (list ℕ) [10]
